src/main/java/apoc/util/QueueBasedSpliterator.java

```diff
@@ -6,49 +6,35 @@
 import java.util.concurrent.BlockingQueue;
 import java.util.function.Consumer;
 
-import static java.util.concurrent.TimeUnit.SECONDS;
-
 /**
  * @author mh
  * @since 06.12.17
  */
 public class QueueBasedSpliterator<T> implements Spliterator<T> {
     private final BlockingQueue<T> queue;
     private T tombstone;
-    private T entry;
     private TerminationGuard terminationGuard;
-    private final long timeout;
+    private boolean foundTombstone = false;
 
     public QueueBasedSpliterator(BlockingQueue<T> queue, T tombstone, TerminationGuard terminationGuard) {
-        this(queue, tombstone, terminationGuard, 10);
-    }
-
-    public QueueBasedSpliterator(BlockingQueue<T> queue, T tombstone, TerminationGuard terminationGuard, long timeout) {
         this.queue = queue;
         this.tombstone = tombstone;
         this.terminationGuard = terminationGuard;
-        this.timeout = timeout;
-        entry = poll();
     }
 
     @Override
     public boolean tryAdvance(Consumer<? super T> action) {
-        if (Util.transactionIsTerminated(terminationGuard)) return false;
-        if (isEnd()) return false;
-        action.accept(entry);
-        entry = poll();
-        return !isEnd();
-    }
-
-    private boolean isEnd() {
-        return entry == null || entry == tombstone;
```

Comment:
afaik there were some places which indicated null for end of (partial-) processing, but I can't remember exactly.